src/services/user/handlers.ts

```diff
@@ -0,0 +1,129 @@
+import type { RequestHandler } from "express";
+import { userSchemas } from "./schemas";
+import { responseMessages } from "@/lib/utils";
+import { PrismaClientKnownRequestError, prisma } from "@/lib/prisma";
+
+const postUser: RequestHandler = async (req, res, next) => {
+  try {
+    if (!req.token)
+      return res.status(401).json({
+        message: responseMessages.error.unAuthorized,
+      });
+
+    const inputBody = userSchemas.postBody.safeParse(req.body);
+    if (!inputBody.success) {
+      return res.status(400).json({
+        message: responseMessages.error.reqBody,
+      });
+    }
+
+    await prisma.user.create({
+      data: {
+        uid: req.token?.decoded.uid,
```

Comment:
no need for optional chaining (`a?.b`). Use regular chaining instead (`a.b`)

Reply:
roger